jetstream/pull.go

```diff
@@ -203,7 +208,12 @@ func (p *pullConsumer) Consume(handler MessageHandler, opts ...PullConsumeOpt) (
 			if msgErr == nil {
 				return
 			}
-			if err := sub.handleStatusMsg(msg, msgErr); err != nil {
+
+			sub.Lock()
+			err := sub.handleStatusMsg(msg, msgErr)
```

Comment:
Can't put the lock inside the `handleStatusMsg` because it is used in `Next` which is globally locked.

Reply:
Yeah, unfortunately `Next()` has to be locked as a whole, which is a bit painful considering the logic of handling `pendingMsgs` and status messages is common across `Consume` and `Next` - I'll probably clean that up for the next release, but for now that will work.

Reply:
A potential trick (to not refactor the whole logic) is to use `TryLock` but the go team does not advise to use it because it means something is probably wrong. 

https://pkg.go.dev/sync#Mutex.TryLock
. Note that while correct uses of TryLock do exist, they are rare, and use of TryLock is often a sign of a deeper problem in a particular use of mutexes.